### PR TITLE
Create sdf world scene

### PR DESCRIPTION
**Pull request by gustavoneves12**
create and manage the sdf world scene in the C/C++ side.

### CODE REVIEW

File: src/Vizkit3dWorld.cpp [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+        //close all openned windows
+        qApp->closeAllWindows();
+        //the correct way is to use guiThread.join()
+        //but if showGui is "false" then the QApplication::exec is frozen
+        //to solve this problem join the thread for 500
+        guiThread.try_join_for(boost::chrono::milliseconds(500));
+    }
+}
+
+void Vizkit3dWorld::wait(){
```
Comment (doudou) [inline src/Vizkit3dWorld.cpp]: The method name really should say what we're waiting for ....

File: src/Vizkit3dWorld.cpp [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+void Vizkit3dWorld::applyTransformation(std::string targetFrame, std::string sourceFrame, base::Position position, base::Orientation orientation) {
+    applyTransformation(targetFrame ,sourceFrame,
+                        QVector3D(position.x(), position.y(), position.z()),
+                        QQuaternion(orientation.w(), orientation.x(), orientation.y(), orientation.z()));
+}
+
+void Vizkit3dWorld::applyTransformation(std::string targetFrame, std::string sourceFrame, QVector3D position, QQuaternion orientation) {
+
+    if (widget) {
+        widget->setTransformation(targetFrame.c_str(),
```
Comment (doudou) [inline src/Vizkit3dWorld.cpp]: Use QString::fromStdString

**Comment by gustavoneves12** [comment]
@doudou  PING

File: src/Utils.hpp [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+/**
+ * Copy image pixels from QImage to base::samples::frame::Frame
+ *
+ * @param src: the source image
+ * @param dst: the destionation image
+ * @param flipImage: if is true, flip image in vertical direction
+ */
+inline void cpyQImageToFrame(const QImage& src, base::samples::frame::Frame& dst, bool flipImage = false)
```
Comment (doudou) [inline src/Utils.hpp:71]: There are helpers for this kind of operations in perception/frame_helpers. You should have a look there.

File: src/Utils.hpp [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+        break;
+    }
+}
+
+/**
+ * Change the thread priority
+ * This functions is used to change the thread priority to max
+ * Is used in this program to set Qt event loop thread to max priority
+ */
+inline void setthread_priority_max(boost::thread& thread){
```
Comment (doudou) [inline src/Utils.hpp:146]: Not used anymore (and should _not_ be used). Please remove.

**Comment by doudou** [comment]
I merged, but you should really look into the frame_helpers package.

**Comment by gustavoneves12** [comment]
@doudou, I looked into frame_helpers, but I did not find a method to convert QImage to base::samples::Frame. I really look, but the version in the rock master only has method to convert base::samples::Frame to QImage or cv::Mat.

**Comment by doudou** [comment]
Alright, we should think of moving your code there, then.

**Pull request merged**